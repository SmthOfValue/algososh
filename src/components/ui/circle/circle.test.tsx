import React from "react";
import renderer from 'react-test-renderer';
import { ElementStates } from "../../../types/element-states";

import { Circle } from './circle';

describe('Рендер Circle', () => {
  it('Circle без букв рендерится без ошибок', () => {
    const circleWithoutLetter = renderer
      .create(<Circle />)
      .toJSON();
      expect(circleWithoutLetter).toMatchSnapshot();
  }); 

  it('Circle с буквами рендерится без ошибок', () => {
    const circleWithLetter = renderer
      .create(<Circle letter='буквы' />)
      .toJSON();
      expect(circleWithLetter).toMatchSnapshot();
  }); 

  it('Circle с head рендерится без ошибок', () => {
    const circleWithHead = renderer
      .create(<Circle head = 'head' />)
      .toJSON();
      expect(circleWithHead).toMatchSnapshot();
  });

  it('Circle с React-элементом в head рендерится без ошибок', () => {

    const reactElement = React.createElement('div');

    const circleWithReactElementAsHead = renderer
      .create(<Circle head = {reactElement} />)
      .toJSON();
      expect(circleWithReactElementAsHead).toMatchSnapshot();
  });

  it('Circle с tail рендерится без ошибок', () => {
    const circleWithTail = renderer
      .create(<Circle tail = 'tail' />)
      .toJSON();
      expect(circleWithTail).toMatchSnapshot();
  });

  it('Circle с React-элементом в tail рендерится без ошибок', () => {

    const reactElement = React.createElement('div');

    const circleWithReactElementAsTail = renderer
      .create(<Circle tail = {reactElement} />)
      .toJSON();
      expect(circleWithReactElementAsTail).toMatchSnapshot();
  });

  it('Circle с index рендерится без ошибок', () => {
    const circleWithIndex = renderer
      .create(<Circle index = {0} />)
      .toJSON();
      expect(circleWithIndex).toMatchSnapshot();
  });

  it('Circle с пропом isSmall = true рендерится без ошибок', () => {
    const circleWithIsSmall = renderer
      .create(<Circle isSmall = {true} />)
      .toJSON();
      expect(circleWithIsSmall).toMatchSnapshot();
  });

  it('Circle в состоянии default рендерится без ошибок', () => {
    const circleInDefaultState = renderer
      .create(<Circle state = {ElementStates.Default} />)
      .toJSON();
      expect(circleInDefaultState).toMatchSnapshot();
  });

  it('Circle в состоянии changing рендерится без ошибок', () => {
    const circleInChangingState = renderer
      .create(<Circle state = {ElementStates.Changing} />)
      .toJSON();
      expect(circleInChangingState).toMatchSnapshot();
  });

  it('Circle в состоянии modified рендерится без ошибок', () => {
    const circleInModifiedState = renderer
      .create(<Circle state = {ElementStates.Modified} />)
      .toJSON();
      expect(circleInModifiedState).toMatchSnapshot();
  });
});