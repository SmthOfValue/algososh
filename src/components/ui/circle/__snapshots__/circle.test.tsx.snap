// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер Circle Circle без букв рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер Circle Circle в состоянии changing рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер Circle Circle в состоянии default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер Circle Circle в состоянии modified рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер Circle Circle с React-элементом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div />
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер Circle Circle с React-элементом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div />
  </div>
</div>
`;

exports[`Рендер Circle Circle с head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер Circle Circle с index рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Рендер Circle Circle с tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Рендер Circle Circle с буквами рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      буквы
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер Circle Circle с пропом isSmall = true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
